### ISSUE TITLE
pyodide.mountNativeFS and syncfs

### ISSUE THREAD

**Issue by gabrielfougeron** [pyodide/pyodide#3456]
## 🐛 Bug

I'm testing out one of the latest 0.22 features: the ability to mount a directory Handle from the File System Access API.

I've successfully monted a handle in pyodide using `NativeFS = await pyodide.mountNativeFS("/mount_dir", dirHandle)` and accessed files from python.
I've written files on the python side and used `await NativeFS.syncfs()` to dump the changes to disk.
These work great.

Now I want to do the opposite: suppose I've made changes on disk, I want them reflected in the pyodide FS.
Looking at the [Emscripten documentation](https://emscripten.org/docs/api_reference/Filesystem-API.html), it should be as easy as calling `await NativeFS.syncfs(populate = false)`.
However, this does not work.

Am I correct in thinking this might be a bug? Maybe you guys simply haven't had time to work on it yet ...
Should I be tackling my problem differently?

Thanks for everything,

Gabriel

### Environment

- Pyodide Version 0.22
- Browser version: Latest Chrome

**Comment by hoodmane** [comment]
The `syncfs` argument doesn't seem to take an argument:
https://github.com/pyodide/pyodide/blob/main/src/js/api.ts?plain=1#L426
This could probably be improved. To sync the other way:
```js
async function syncFromDisk() {
	return await new Promise((resolve, _) => pyodide.FS.syncfs(true, resolve));
}
```

**Comment by ryanking13** [comment]
Thanks for the report @gabrielfougeron.

> Am I correct in thinking this might be a bug? Maybe you guys simply haven't had time to work on it yet ...
Should I be tackling my problem differently?

Actually, I naively thought that people would want the "dump the changes to disk" case only. But yeah, I agree that the opposite should be possible.

The implementation of `syncfs` is fairly simple, so would you please open a PR that adds a `populate` parameter to it?

https://github.com/pyodide/pyodide/blob/eb7b80bb3e12b9e481e0974a30f4fb6a126cff22/src/js/api.ts#L426-L428

**Comment by hoodmane** [comment]
I don't really like the populate parameter. Why not have two different functions `syncToDisk` and `syncFromDisk` or something?

**Comment by ryanking13** [comment]
> I don't really like the populate parameter. Why not have two different functions syncToDisk and syncFromDisk or something?

Yeah, that's a good point. The populate parameter is not very intuitive... I always search Emscripten docs whether I should use false or true.

I'm +1 with having two different functions, and probably deprecating the name `syncfs`.
